### ISSUE TITLE
Accessibility failing on Lighthouse

### ISSUE THREAD

**Issue by devcer** [jaredLunde/masonic#133]
**Describe the bug**
Accessibility of masonic is failing. Most likely because `role="gridcell"` is not a child of `role="row"`, instead it is a child of `role="grid"`

**To Reproduce**

1. Go to example sandbox -> https://codesandbox.io/s/0oyxozv75v
2. Open the preview in new window
3. Open Devtools and click on Lighthouse
4. Now select only accessibility test in desktop view and run lighthouse
5. Check the analysis and you will see two accessibility errors.

**Expected behavior**
Masonic should be accessible

**Screenshots**
<img width="1440" alt="Screenshot 2022-09-22 at 8 12 42 AM" src="https://user-images.githubusercontent.com/3799600/191647528-8fd69f65-2f4b-48a0-9379-6a4c5729ff4f.png">

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

- Device: Macbook Pro
- OS: v12.6
- Browser -  Chrome
- Version 105.0.5195.125 

**Additional context**

Elements that have role="gridcell" applied to them must be the child of an element with a role of [row](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/row_role). -> https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/gridcell_role

https://dequeuniversity.com/rules/axe/4.4/aria-required-parent

**Comment by jaredLunde** [comment]
PRs welcome :)

**Comment by V3RON** [comment]
I wonder if we should remove the `role` prop altogether and force `list` and `listitem` roles. Technically, the grid created by Masonic doesn't reassemble a table at all, but rather a list with styles applied to mimic a tabular layout. 🤔
